tardis/io/util.py

```diff
@@ -138,6 +138,57 @@ def yaml_load_file(filename, loader=yaml.Loader):
         return yaml.load(stream, Loader=loader)
 
 
+def parse_species_list(sdec_plotter, data, species_list, packets_mode, nelements=None):
+        """
+        Parse user requested species list and create list of species ids to be used.
+
+        Parameters
+        ----------
+        species_list : list of species to plot
+            List of species (e.g. Si II, Ca II, etc.) that the user wants to show as unique colours.
+            Species can be given as an ion (e.g. Si II), an element (e.g. Si), a range of ions
+            (e.g. Si I - V), or any combination of these (e.g. species_list = [Si II, Fe I-V, Ca])
+        packets_mode : str, optional
+            Packet mode, either 'virtual' or 'real'. Default is 'virtual'.
+        nelements : int, optional
+            Number of elements to include in plot. The most interacting elements are included. If None, displays all elements.
+
+        Raises
+        ------
+        ValueError
+            If species list contains invalid entries.
+
+        """
+        from tardis.util.base import atomic_number2element_symbol
```

Comment:
why is this import defined here?

Reply:
It is defined inside the function to break the circular dependency between tardis.io.util and tardis.util.base

Reply:
and in the below sdec_plot
The parse_species_list function in util.py builds on the sdec_plot.py version by not only parsing species but also filtering them based on interaction frequency in the data. This makes it more data-driven and adaptable for cases where top interacting elements are needed.

Reply:
Imports should be defined at the start of the file right?

Reply:
Yes, but importing this statement at the start of the file creates a circular dependency. Specifically, atomic_number2element_symbol is imported from tardis.util.base, and within tardis.util.base, get_internal_data_path is imported from util.py, which triggers a circular loop. Refactoring the functions to avoid this would increase code complexity and could introduce new errors.